dev-support/jenkins.sh

```diff
@@ -114,22 +114,47 @@ function check_ci_run() {
 function run_ci() {
   TESTPATCHBIN="${WORKSPACE}/${YETUS}/precommit/src/main/shell/test-patch.sh"
 
-  # this must be clean for every run
-  if [[ -d "${PATCHDIR}" ]]; then
-    rm -rf "${PATCHDIR:?}"
+  if [[ "$IS_WINDOWS" && "$IS_WINDOWS" == 1 ]]; then
+    echo "Building in a Windows environment, skipping some Yetus related settings"
+  else
+    # run in docker mode and specifically point to our
+    # Dockerfile since we don't want to use the auto-pulled version.
+    YETUS_ARGS+=("--docker")
+    YETUS_ARGS+=("--dockerfile=${DOCKERFILE}")
+    YETUS_ARGS+=("--mvn-custom-repos")
+    YETUS_ARGS+=("--dockermemlimit=22g")
+
+    # test with Java 8 and 11
+    YETUS_ARGS+=("--java-home=/usr/lib/jvm/java-8-openjdk-amd64")
+    YETUS_ARGS+=("--multijdkdirs=/usr/lib/jvm/java-11-openjdk-amd64")
+    YETUS_ARGS+=("--multijdktests=compile")
   fi
-  mkdir -p "${PATCHDIR}"
-
-  # if given a JIRA issue, process it. If CHANGE_URL is set
-  # (e.g., Github Branch Source plugin), process it.
-  # otherwise exit, because we don't want Hadoop to do a
-  # full build.  We wouldn't normally do this check for smaller
-  # projects. :)
-  if [[ -n "${JIRA_ISSUE_KEY}" ]]; then
-    YETUS_ARGS+=("${JIRA_ISSUE_KEY}")
-  elif [[ -z "${CHANGE_URL}" ]]; then
-    echo "Full build skipped" >"${PATCHDIR}/report.html"
-    exit 0
+
+  if [[ "$IS_NIGHTLY_BUILD" && "$IS_NIGHTLY_BUILD" == 1 ]]; then
+    YETUS_ARGS+=("--empty-patch")
+    YETUS_ARGS+=("--branch=${BRANCH_NAME}")
```

Comment:
Where was this before?

Reply:
The Jenkins nightly CI for Linux defines these flags in the Jenkins UI itself. Infact, it runs a shell script for the nightly CI - https://ci-hadoop.apache.org/view/Hadoop/job/hadoop-qbt-trunk-java8-linux-x86_64/configure.

I don't like the idea of having flags at different places (some in Jenkins UI and some in shell scripts). Thus, I've added all of them here.